### PR TITLE
Reclaim Setup

### PR DESCRIPTION
**Pull request by Will-Cooper**
The details about the changes made to get reclaim working (Apache/ jelastic/ flask).

### CODE REVIEW

**Comment by Will-Cooper** [comment]
That path stuff was a real nuisance but now at least the tests work/ the site runs locally/ the site runs on reclaim

**Comment by Will-Cooper** [comment]
This repo `main` branch is out of sync, but the new `reclaim` branch isn't as I've made the same changes in both places. That was manually though so I'd prefer to merge, then do a git pull in reclaim, to be sure it's alright.

**Comment by kelle** [comment]
I think of rootpath as my path to "root" ...the upper most directory. But it's used here to mean the path to the simple app code. Consider renaming this file to simple_path

File: README.md [hunk 1]
```diff
@@ -50,6 +50,30 @@ If updating requirements, update them with:
...
+Alias /static <your/path/to/root>/simple_app/static
+Alias /favicon.ico <your/path/to/root>/simple_app/static/favicon.ico
+```
+Finally, in the `rootpath` symlink, swap
+`
+.
+`
+for
+`
+<your/path/to/root>
```
Comment (kelle) [inline README.md:72]:
```suggestion
<your/path/to/simple>
```

File: simple_app.wsgi [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#/usr/bin/python
+import sys
+sys.path.insert(0, "rootpath/simple_app")
+sys.path.insert(0, "rootpath")
```
Comment (kelle) [inline simple_app.wsgi:4]:
```suggestion
sys.path.insert(0, "simple_path/simple_app")
sys.path.insert(0, "simple_path")
```

File: simple_app/app_simple.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 This is the main script to be run from the directory root, it will start the Flask application running which one can
 then connect to.
 """
+import sys
+sys.path.append('rootpath/simple_app')
```
Comment (kelle) [inline simple_app/app_simple.py:6]:
```suggestion
sys.path.append('simple_path/simple_app')
```

**Review by kelle — CHANGES_REQUESTED** [review]
Excellent progress!!

**Comment by kelle** [comment]
`simple_root` sounds good to me!

File: simple_app/utils.py [hunk 4]
```diff
@@ -4,8 +4,8 @@
 import sys
 # local packages
 
-sys.path.append('.')
-from simple_app.simports import *
+sys.path.append('rootpath/simple_app')
```
Comment (Will-Cooper) [inline simple_app/utils.py:7]:
```suggestion
sys.path.append('simple_root/simple_app')
```

File: simple_app/utils.py [hunk 5]
```diff
@@ -23,7 +23,7 @@ def sysargs():
     _args.add_argument('-p', '--port', default=8000,
                        help='Local port number to host server through, default 8000', type=int)
     _args.add_argument('-d', '--debug', help='Run Flask in debug mode?', default=False, action='store_true')
-    _args.add_argument('-f', '--file', default='SIMPLE.db',
+    _args.add_argument('-f', '--file', default='rootpath/SIMPLE.db',
```
Comment (Will-Cooper) [inline simple_app/utils.py:26]:
```suggestion
    _args.add_argument('-f', '--file', default='simple_root/SIMPLE.db',
```

File: simple_app/utils.py [hunk 6]
```diff
@@ -229,7 +229,7 @@ class JSCallbacks:
 
     def __init__(self):
         jsfuncnames = ('dropdownx_js', 'dropdowny_js', 'button_flip', 'normslider', 'reset_slider')
-        with open('simple_app/simple_callbacks.js', 'r') as fcall:
+        with open('rootpath/simple_app/simple_callbacks.js', 'r') as fcall:
```
Comment (Will-Cooper) [inline simple_app/utils.py:232]:
```suggestion
        with open('simple_root/simple_app/simple_callbacks.js', 'r') as fcall:
```

File: simple_app/tests/test_utils.py [hunk 7]
```diff
@@ -7,18 +7,20 @@
...
+db_name = 'rootpath/temp.db'
+db_cs = f'sqlite:///{db_name}'
 
 
 @pytest.fixture(scope='module')
 def db():
     if os.path.exists(db_name):
         os.remove(db_name)
-    copy('SIMPLE.db', db_name)
+    copy('rootpath/SIMPLE.db', db_name)
```
Comment (Will-Cooper) [inline simple_app/tests/test_utils.py:23]:
```suggestion
    copy('simple_root/SIMPLE.db', db_name)
```

File: simple_app/tests/test_utils.py [hunk 8]
```diff
@@ -7,18 +7,20 @@
...
 """
+import sys
+sys.path.append('rootpath/simple_app')
 # local packages
-from simple_app.utils import *
+from utils import *
 
-db_name = 'temp.db'
-db_cs = 'sqlite:///temp.db'
+db_name = 'rootpath/temp.db'
```
Comment (Will-Cooper) [inline simple_app/tests/test_utils.py:15]:
```suggestion
db_name = 'simple_root/temp.db'
```

File: simple_app/tests/test_utils.py [hunk 9]
```diff
@@ -7,18 +7,20 @@
 absmags -- called by results_concat
 coordinate_project -- called by results_concat
 """
+import sys
+sys.path.append('rootpath/simple_app')
```
Comment (Will-Cooper) [inline simple_app/tests/test_utils.py:11]:
```suggestion
sys.path.append('simple_root/simple_app')
```

File: simple_app/tests/test_plots.py [hunk 10]
```diff
@@ -2,15 +2,17 @@
 Testing the plot functions
 """
 # local packages
-from simple_app.plots import *
-from simple_app.tests.test_utils import *
+import sys
+sys.path.append('rootpath/simple_app')
```
Comment (Will-Cooper) [inline simple_app/tests/test_plots.py:6]:
```suggestion
sys.path.append('simple_root/simple_app')
```

File: simple_app/plots.py [hunk 11]
```diff
@@ -2,11 +2,10 @@
 File containing the 'workhorse' functions generating the various plots seen on the website
 """
 import sys
-# local packages
-import numpy as np
+sys.path.append('rootpath/simple_app')
```
Comment (Will-Cooper) [inline simple_app/plots.py:5]:
```suggestion
sys.path.append('simple_root/simple_app')
```

File: simple_app/app_simple.py (hunk 3 above)
Comment (Will-Cooper) [inline simple_app/app_simple.py:6]:
```suggestion
sys.path.append('simple_root/simple_app')
```

File: README.md [hunk 12]
```diff
@@ -50,6 +50,30 @@ If updating requirements, update them with:
...
+```
+WSGIApplicationGroup %{GLOBAL}
+```
+Additionally, edit `wsgi.conf` with the lines:
+```
+WSGIScriptAlias / <your/path/to/root>/simple_app.wsgi
+Alias /static <your/path/to/root>/simple_app/static
+Alias /favicon.ico <your/path/to/root>/simple_app/static/favicon.ico
+```
+Finally, in the `rootpath` symlink, swap
```
Comment (Will-Cooper) [inline README.md:66]:
```suggestion
Finally, in the `simple_root` symlink, swap
```

File: simple_app.wsgi (hunk 2 above)
Comment (Will-Cooper) [inline simple_app.wsgi:4]:
```suggestion
sys.path.insert(0, "simple_root/simple_app")
sys.path.insert(0, "simple_root")
```

**Review by dr-rodriguez — APPROVED** [review]

**Review by kelle — APPROVED** [review]
This looks great!  Thanks for renaming the "root" directory. Reads much more clearly to me now.

**Pull request merged**